### PR TITLE
Express http

### PR DESCRIPTION
**Pull request by colleend**
Checks for usage of http servers instead of https servers. Not sure if this is actually a good rule, because it may give too many false positives. I wrote it without thinking about the FP rate, so if its not useful we can discard.

### CODE REVIEW

**Comment by colleend** [comment]
not sure if this check is actually good -- there might be high FP rate, and some app-writers might also want to serve http but redirect all requests to https.

**Comment by minusworld** [comment]
Yeah, we may run into some issues like that occasionally. Since we'll be shipping it in packs, ostensibly they can pick-and-choose specific things for their use-case. Hopefully that mitigates the issue.

If you're interested in spot-checking how the rule does on real code, consider giving it a spin on platform! dev.massive.ret2.co

File: problem-based-packs/insecure-transport/js-express/using-http-server.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    - pattern: |
+        http.$FUNC(...);
+    - pattern: |
+        new http.$FUNC2(...);
+  message: |
+    Checks for any usage of http servers instead of https servers. Encourages the usage of https protocol instead of 
+    http, which does not have TLS and is therefore unencrypted. Using http can lead to man-in-the-middle attacks 
+    in which the attacker is able to read sensitive information.
+  metadata:
+    vulnerability: Insecure transport
```
Comment (minusworld) [inline problem-based-packs/insecure-transport/js-express/using-http-server.yaml]: This is probably OWASP 2017 Top 10: A3 Sensitive Data Exposure.
Would you also be able to add in the CWE? You can just Google "CWE http" or something simlar.

Comment (colleend) [reply problem-based-packs/insecure-transport/js-express/using-http-server.yaml]: yes definitely!!

**Comment by colleend** [comment]
> Yeah, we may run into some issues like that occasionally. Since we'll be shipping it in packs, ostensibly they can pick-and-choose specific things for their use-case. Hopefully that mitigates the issue.
> 
> If you're interested in spot-checking how the rule does on real code, consider giving it a spin on platform! dev.massive.ret2.co

seems like there's not too many false positive rates after scanning!

**Pull request merged**